fuel-vm/src/tests/wideint.rs

```diff
@@ -96,6 +96,107 @@ fn cmp_u128(
     }
 }
 
+#[test]
+fn cmp_u128_resets_of() {
+    // Given
+    // Issue an overflowing operation first and log the value of $of
+    let mut ops = Vec::new();
+    ops.push(op::movi(0x20, Flags::WRAPPING.bits() as u32));
+    ops.push(op::flag(0x20));
+    ops.extend(make_u128(0x20, u128::MAX));
+    ops.extend(make_u128(0x21, 3u64.into()));
+    ops.extend(make_u128(0x22, 2u64.into()));
+    ops.extend(make_u128(0x23, 0u64.into()));
+    ops.push(op::wdmd(0x23, 0x20, 0x21, 0x22));
```

Comment:
This could be replaced by a simpler set of operations that sets `$of`. For instance,

```rust
op::not(0x20, RegId::ZERO), // set 0x20 to u64::MAX
op::mul(0x20, 0x20, 0x20), // multiply by itself
```

Reply:
Thanks, looks much better. Done in 5016859bc8345e6a461656c3e49e2866a00fd45d